aloha-core/src/main/scala/com/eharmony/aloha/dataset/csv/json/csvJson.scala

```diff
@@ -85,61 +112,125 @@ extends DefaultJsonProtocol
-                case Some("boolean") if optional => o.convertTo[OptionCsvColumnWithDefault[Boolean]]
-
-                case Some("string")  => o.convertTo[CsvColumnWithDefault[String]]
-                case Some("double")  => o.convertTo[CsvColumnWithDefault[Double]]
-                case Some("float")   => o.convertTo[CsvColumnWithDefault[Float]]
-                case Some("long")    => o.convertTo[CsvColumnWithDefault[Long]]
-                case Some("int")     => o.convertTo[CsvColumnWithDefault[Int]]
-                case Some("short")   => o.convertTo[CsvColumnWithDefault[Short]]
-                case Some("byte")    => o.convertTo[CsvColumnWithDefault[Byte]]
-                case Some("char")    => o.convertTo[CsvColumnWithDefault[Char]]
-                case Some("boolean") => o.convertTo[CsvColumnWithDefault[Boolean]]
-
-                case Some("enum") =>
-                    enumClassName.map {
-                        case name if optional =>
-                            val clazz = enumClass getOrElse {
-                                throw new DeserializationException(s"class '$name' is not an Enum class.")
-                            }
-                            o.convertTo(OptionEnumCsvColumn.oeccFormat(clazz))
-                        case _ => o.convertTo[EnumCsvColumn]
-                    } orElse {
-                        if (optional)
-                            Option(o.convertTo[OptionSyntheticEnumCsvColumn])
-                        else Option(o.convertTo[SyntheticEnumCsvColumn])
-                    } getOrElse {
-                        throw new DeserializationException(s"Couldn't create enum type: $o")
-                    }
-
-                case None =>
-                    debug(s"No type provided.  Assuming Any.  Given: ${o.compactPrint}")
-                    o.convertTo[DefaultCsvColumn]
+            implicit val riJf = new StdRefInfoToJsonFormat
+
+            val spec: CsvColumn = (sized, optional) match {
+                case (false, false) =>
+                    scalar(CsvColumnSpecWithDefaultFormat, reqEnum, fieldType, o, enumClassName, enumClass)
+                case (false, true) =>
+                    scalar(OptionCsvColumnSpecWithDefaultFormat, optEnum, fieldType, o, enumClassName, enumClass)
+                case (true, false) =>
+                    vector(SeqCsvColumnSpecWithNoDefaultFormat, fieldType, o)
+                case (true, true) =>
+                    vector(OptionSeqCsvColumnSpecWithNoDefaultFormat, fieldType, o)
             }
+
             spec
         }
     })
+
+    private[json] def convertType(
+            reader: ReaderProducer,
+            aStr: String,
+            o: JsObject)(implicit riJf: RefInfoToJsonFormat): Either[String, CsvColumn] = {
+        for {
+            riUncast <- RefInfo.fromString(aStr.capitalize).right
+            ri <- Right(riUncast.asInstanceOf[RefInfo[Any]]).right // cast :-(
```

Comment:
Should this be wrapped in a `Try` in case the cast fails?

Reply:
Done